---
title: "Project"
author: "Andreas Breuß, Anastassia Fink, Leon Hornich, Yuan Sun"
date: "03.06.2021"
output: html_document
---

1. Install Bioconducter 
#```{r}

if (!requireNamespace("BiocManager", quietly=TRUE))
  install.packages("BiocManager")
BiocManager::install(version = "3.12")

```

2. Install affy, vsn and AnnotationDbi
#```{r}

BiocManager::install("affy")
BiocManager::install("vsn")
BiocManager::install("AnnotationDbi")
install.packages("rstudioapi")
install.packages("tidyverse")
install.packages("plotrix")

```


---
## R Markdown

```{r}
knitr::opts_chunk$set(echo = TRUE)
```

# define working directory
```{r setup, include=FALSE}


wd="C:/Users/Ana/OneDrive - bwedu/Bachelor MoBi/4. Fachsemester/Data Analysis/RStudio"


setwd(paste0(wd,"/tra/"))


```


# libraries

```{r}
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf, lib.loc = "C:/Users/Ana/Documents/Packages")
library(hgu133plus2hsenstprobe, lib.loc ="C:/Users/Ana/Documents/Packages")
library(hexbin)
library(rstudioapi)
library(tidyverse)
library(stringr)
library(readr)
library(dplyr)
library(plotrix)
library(ggplot2)
```



# Read in .CEL files
```{r}

setwd(paste0(wd, "/rawdata/GSE149507 lung cancer"))

data = ReadAffy()
data@cdfName<-"HGU133PLUS2_Hs_ENST"

setwd(paste0(wd,"/sessions/rda"))

lungcancer_matrix <- as.data.frame(exprs(data))

save.image(file="rawdata.lungcancer.pdf")
image(data, col = rainbow(100, start = 0, end = 0.75)[100:1])



```


# overview / will be changed
#```{r}

#microarrays
dim(lungcancer_matrix)[2]

# names of the samples
colnames(data)
substr(colnames(lungcancer_matrix), 1,10)

# gene count
dim(lungcancer_matrix)[1]

# probe set names (or Affymetrix IDs) - are changed when changing cdfName
featureNames(data)[1:10]

#expression table
head(lungcancer_matrix)

```

#```{r}

setwd(paste0(wd,"/plots"))

for (i in 1:24){
  image(data[,i], col = rainbow(100, start = 0, end = 0.75)[100:1])
  file.name = paste0("lungcancer_", str_remove(colnames(lungcancer_matrix)[1],".CEL"),".pdf")
  # dev.copy2pdf(file = file.name)
}

```


# cleanup NA and/or space
```{r}
 
apply(lungcancer_matrix, 2, function(x){sum(is.na(x))})
apply(lungcancer_matrix, 2, function(x){sum(x == 0)})
sum(sapply(rownames(lungcancer_matrix), function(x){sum(x == "")}))


```

```{r}

data.norm <- vsnrma(data)
data_vsnrma_matrix <- exprs(data.norm)

head(data.norm)
head(data_vsnrma_matrix)

#save

setwd(paste0(wd,"/sessions/rda"))

save.image(file="normalized_data..pdf")

```


# Plots
```{r}

#meanSdPlot
meanSdPlot(data.norm)

setwd(paste0(wd,"/plots/"))
dev.copy2eps(file="meanSdPlot_lungcancer_vsnrma_normalized.eps")



# Boxplot before normalization

par(las = 2) #axis labels 

par(mai = c(0.5,1,0.5,0.1))

boxplot(data, names = substr(colnames(lungcancer_matrix), 1,12),
        col = rainbow(24), cex.axis=0.6, 
        main="sample expression Lungcancer", 
        horizontal = FALSE)


setwd(paste0(wd,"/plots/"))
dev.copy2eps(file="boxplot_lungcancer_rawdata.pdf")



#Boxplot normalized data

par(las = 2) #axis labels 

par(mai = c(0.5,1,0.5,0.1))

boxplot(data_vsnrma_matrix, names = substr(colnames(lungcancer_matrix), 1,12),
        col = rainbow(24), cex.axis=0.6, 
        main="sample expression Lungcancer", 
        horizontal = FALSE)


setwd(paste0(wd,"/plots/"))
dev.copy2eps(file="boxplot_lungcancer_normalized.pdf")


```



# RNA degeneration plot
```{r}

rnadeg.raw = AffyRNAdeg(data)

#shift & scale
plotAffyRNAdeg(rnadeg.raw, col=rainbow(24), transform = "shift.scale")
title(sub = "lungcancer rawdata")


setwd(paste0(wd, "/plots/" ,sep = "/"))
dev.copy2pdf(file = "lungcancer_rnadeg_rawdata.pdf")


#shift
plotAffyRNAdeg(rnadeg.raw, col = rainbow(24), transform = "shift.only")
title(sub = "lungcancer rawdata")


setwd(paste0(wd, "/plots/" ,sep = "/"))
dev.copy2pdf(file = "lungcancer_rawdata.pdf")

```



# Scatterplot

```{r}

setwd(paste0(wd,"/plots/"))

for (i in 1:11){
  plot(exprs(data.norm)[,i], exprs(data.norm)[,i+1], pch = ".",
       xlab = substr(colnames(lungcancer_matrix), 1,10)[i],
       ylab = substr(colnames(lungcancer_matrix), 1,10)[i+1])
  abline(0,1,col="red")
  
  title(main = paste("Scatterplot of probe", 
                     substr(colnames(lungcancer_matrix), 1,10)[i],
                     "and",
                     substr(colnames(lungcancer_matrix), 1,10)[i+1],
                     sep = " ", collapse = NULL))
  
  file.name = paste("diabetes_scatterplot_",
                    substr(colnames(lungcancer_matrix), 1,10)[i],
                     "_",
                    substr(colnames(lungcancer_matrix), 1,10)[i+1],
                    ".pdf",
                    sep = "")
  dev.copy2pdf(file = file.name)
}


```

# Extract expression values
```{r}

dataExprs = exprs(data.norm)
head(dataExprs)
dim(dataExprs)

```

# ensemble IDs
```{r}

dataExprs = dataExprs[grepl("ENST", rownames(dataExprs)), ]
head(dataExprs)
dim(dataExprs)

```

# remove x_at suffix
```{r}

rownames(dataExprs) <- unlist(lapply(strsplit(rownames(dataExprs), split = "\\."), "[", 1))
head(dataExprs)

```

# ensemble IDs and gene symbols
```{r}

setwd(paste0(wd, "/tables"))
ensembl_df = read.csv("ensembl.103.txt", sep = ",")
head(ensembl_df)

transcriptIDs = as.character(ensembl_df[, "Transcript.stable.ID"])
geneSymbol = as.character(ensembl_df[, "HGNC.symbol"])
names(geneSymbol) = transcriptIDs
head(geneSymbol)

```

# Replace ensemble IDs with HGNC symbol
```{r}

chipIDs = rownames(dataExprs)
noMatchIDs = chipIDs[!chipIDs %in% transcriptIDs]
length(noMatchIDs)

newChipIDs = chipIDs[chipIDs %in% transcriptIDs]
dataExprs = dataExprs[newChipIDs, ]
dim(dataExprs)

symbol = geneSymbol[newChipIDs]
rownames(dataExprs) = as.character(symbol)

head(dataExprs)

```

# TRA tissues / genesofinterest
```{r}

setwd(paste0(wd,"/tra/"))

human1 =read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv",sep=",")
human2 =read.csv(file="tra.2014.human.5x.table.tsv",sep="\t")
human3 =read.csv(file="tra.2014.human.roth.5x.table.tsv",sep="\t")
human4 =read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")


TRA1.symbol=human1[,6]
TRA2.symbol=human2[,3]
TRA3.symbol=human3[,3]
TRA4.symbol=human4[,3]


CSN1_tissue = human1[grep("CSN", TRA1.symbol),11]
CSN1_symbol = human1[grep("CSN", TRA1.symbol),6]
CSN2_tissue = human2[grep("CSN", TRA2.symbol),11]
CSN2_symbol = human2[grep("CSN", TRA2.symbol),3]
CSN3_tissue = human3[grep("CSN", TRA3.symbol),11]
CSN3_symbol = human3[grep("CSN", TRA3.symbol),3]
CSN4_tissue = human4[grep("CSN", TRA4.symbol),10]
CSN4_symbol = human4[grep("CSN", TRA4.symbol),3]


tissue_vector <- c(CSN1_tissue, CSN2_tissue,CSN3_tissue,CSN4_tissue)
tissue_df <- data.frame(tissue_vector)
symbol_vector <- c(CSN1_symbol, CSN2_symbol,CSN3_symbol,CSN4_symbol)
tissue_df["symbol"] <- c(symbol_vector)

tissue_df <- tissue_df %>% mutate(tissue_vector = str_replace(string = tissue_vector, pattern = "Testis",replacement = "testis"))
tissue_df <- tissue_df %>% mutate(tissue_vector = str_replace(string = tissue_vector, pattern = "testes",replacement = "testis"))
tissue_df <- tissue_df %>% mutate(tissue_vector = str_replace(string = tissue_vector, pattern = "Minor Salivary Gland",replacement = "salivary_gland"))

tissue_df <- tissue_df[!duplicated(tissue_df), ]


tissuecount = data.frame(table(tissue_df[,"tissue_vector"]))


id <- c('A','B','C','D','E','F','G','H','I','J','K')
tissuecount["index"] <- id
 

#dplyr::mutate(id = LETTERS[row_number()] )
```
pl <- ggplot(data = tissuecount, aes(x= '',y = tissuecount[,"Freq"], fill = paste0(id,' : ',Var1,'(', round(tissuecount[,"Freq"]/sum(tissuecount[,"Freq"])*100),'%)' )))

pl <- pl + geom_bar(width = 1, stat = "identity")
pl <- pl + geom_text(aes(x = 1.4,label = id ), position = position_stack(vjust = 0.5))
pl <- pl + theme_void()
pl <- pl + theme_classic()
pl <- pl + theme(legend.position = "top")
pl  <- pl + coord_polar("y", start=0)
pl <- pl + scale_fill_manual(values = palette)
pl <- pl +   theme(axis.line = element_blank())
pl <- pl +   theme(axis.text = element_blank())
pl <- pl +   theme(axis.ticks = element_blank())
pl <- pl +   labs(x = NULL, y = NULL, fill = NULL)
pl <- pl + labs(title ="Pie chart of tissue distribution of CSN TRA´s")
pl <- pl + labs(subtitle ="all CSN genes found in TRA list")

pl
```{r}

pl <- pie3D(tissuecount[,"Freq"],labels=rownames(tissuecount[,"Freq"]),explode=0.1, main="Tissue distribution - TRA´s ")



pl

```

# TRA Chromosomes

```{r}

CSN1 = grep("CSN", TRA1.symbol)
CSN2 = grep("CSN", TRA2.symbol)
CSN3 = grep("CSN", TRA3.symbol)
CSN4 = grep("CSN", TRA4.symbol)


chrom1 = human1[CSN1,"Chromosome"]
chrom2 = human2[CSN2,7]
chrom3 = human3[CSN3,7]
chrom4 = human4[CSN4,"ensembl.chrom"]

chrom <- c(chrom1,chrom2,chrom3,chrom4)
chrom_df = data.frame(chrom)
chrom_df["symbols"] <- c(human1[CSN1,6],human2[CSN2,3],human3[CSN3,3],human4[CSN4,3] )

chrom_df <-chrom_df[!duplicated(chrom_df), ]

chrom = chrom_df[,"chrom"]

#plot


barplot(table(chrom),
main="Chromosomes of human CSN-genes",
xlab="Chromosome",
ylab="Count",
col = rainbow(24)
)


```

# heatmap of cancersets

```{r}

genelist = rownames(dataExprs)

CSN = grep("CSN", genelist)
ELF5 = grep("ELF5", genelist)
LALBA = grep("LALBA", genelist)
SULT = grep("SULT", genelist)
ODAM = grep("ODAM", genelist)

genesofinterest <- c(CSN, ELF5,LALBA,SULT,ODAM)

heatmap(dataExprs[genesofinterest,])


```




Time for the PCA

```{r}
#PCA 

data_csn <- dataExprs[CSN, ]  # get the data for CSN genes from the expression data frame
csn_transcripts <- rownames(as.data.frame(data_csn))
rownames(data_csn) <- csn_transcripts



data_pca <- t(data_csn)



```


```{r}
#Data filtering
library(ggcorrplot)
pca.corr <- cor(data_pca) # correlation matrix

ggcorrplot(pca.corr) # correlation plot before filtering



```

```{r}

variables_filter <- rev(rownames(pca.corr))
 for (var in variables_filter) {
  tmp <- pca.corr[var,-which(colnames(pca.corr) == var)]
  if (any(tmp > 0.8)) {
    pca.corr <- pca.corr[-which(rownames(pca.corr) == var), -which(colnames(pca.corr) == var)]
  }
}

ggcorrplot(pca.corr) # correlation plot after filtering highly correlated variables



```


```{r}

csn_pca_names <- rownames(pca.corr) # The names of the transcripts with a correlation below 0.8


filtered_csn <- subset(dataExprs, rownames(dataExprs) %in% csn_pca_names) #expression data of the filtered names

csn_filtered_transcripts <- rownames(as.data.frame(filtered_csn))
rownames(filtered_csn) <- csn_filtered_transcripts

filtered_pca <- t(filtered_csn)

```



```{r}

res_pca <- prcomp(filtered_pca , scale = TRUE) #perform the pca
summary(res_pca)


```


```{r}
# Screenplot for the first 11 PCs

screeplot(res_pca, type = "l", npcs = 11, main = "Screeplot of the 11 PCs")
abline(h = 1, col="red", lty=5)
legend("topright", legend=c("Eigenvalue = 1"),
       col=c("red"), lty=5, cex=0.6)



```






```{r}
# cumulative variance plot to visualize the PCs with >90% cumulative variance

cumpro <- cumsum(res_pca$sdev^2 / sum(res_pca$sdev^2))
plot(cumpro, xlab = "PC #", ylab = "Amount of explained variance", main = "Cumulative variance plot")
abline(v = 5, col="blue", lty=5)
abline(h = 0.91381, col="blue", lty=5)
legend("topleft", legend=c("Cut-off @ PC1"),
       col=c("blue"), lty=5, cex=0.6)

```

````{r}
# plotting the two PCs which explain the most variance

plot(res_pca$x[,1],res_pca$x[,2], xlab="PC1 (%)", ylab = "PC2 (%)", main = "PC1 / PC2 - plot")



```

```{r}
library(factoextra)
library(stringr)

samples <- rownames(filtered_pca)
group <- sapply(samples, function(sample) {
  ifelse(str_detect(sample, "_ca"), "Cancer", "Healthy")
})

fviz_pca_biplot(res_pca,
             label = "var",
             habillage = group,
             addEllipses = TRUE,
             select.var = list(contrib = 10)) +
  theme_minimal() +
  ggtitle("2D PCA-plot from 181 feature dataset") +
  theme(plot.title = element_text(hjust = 0.5))

```








