---
title: "Project"
author: "Andreas Breu√ü"
date: "27 4 2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown


###```{r}

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.12")


BiocManager::install("affy")

BiocManager::install("vsn")
```



```{r}

library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf, lib.loc = "C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/Packages/")
library(hgu133plus2hsenstprobe, lib.loc ="C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/Packages/")
library(hexbin)

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/tables/")

data.breast <- read.csv(file= "ensembl.103.txt", header = TRUE, sep=",")


setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/rawdata/")



#Kontrolle
#ind = which (new%in%c ()) 
##dev.copy2eps(file="QC_.....eps")
##setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/plots/")
#data.norm=vsnrma(data)

data.breast = ReadAffy()
data.breast@cdfName<-"HGU133PLUS2_Hs_ENST"

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/sessions/rda")

save.image(file="rawdata.breast.rda")
image(data.breast, col = rainbow(100, start = 0, end = 0.75)[100:1])

breast.vsnrma <- vsnrma(data.breast)

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/sessions/rda")

save.image(file="normalized_data.rda")

meanSdPlot(breast.vsnrma)

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/plots/")
dev.copy2eps(file="meanSdPlot_breast_vsnrma_normalized.eps")

# Boxplot

x11(w=10, h=7)
par(las=2)
mmi=c(1,0.7,1.0477939,0.5366749)
par(mai=mmi)

boxplot(data.breast, col=rainbow(150),cex.axis=0.5, main="Gene expression in human breast cancer")

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/plots/")
dev.copy2eps(file="boxplot_breast_rawdata.eps")

#normalized data

x11(w=10, h=7)
par(las=2)
mmi=c(1,0.7,1.0477939,0.5366749)
par(mai=mmi)

boxplot(exprs(breast.vsnrma), col=rainbow(150),cex.axis=0.5, main="Gene expression in human breast cancer")

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/plots/")
dev.copy2eps(file="boxplot_breast_vsnrma_normalized.eps")

#Scatterplot

x11(w=10, h=7)
par(las=2)

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/plots/")

plot(exprs(breast.vsnrma)[,c(1,2)],pch=".")+
abline(0,1,col="red")

dev.copy2eps(file="scatter.plot")
dev.off()

```



#Analysis

setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/tables/")
data.breast <- read.csv(file= "ensembl.103.txt", header = TRUE, sep=",")

ensembl.genes = as.character(data.breast[,1])
ensembl.transcripts = as.character(data.breast[,2])
ensembl.chromosome = as.character(data.breast[,3])
ensembl.symbols = as.character(data.breast[,4])
ensembl.start = as.character(data.breast[,5])

#transcript names

transcript.ids = rownames(data.vsnrma)

tr = substr(transcripts.ids,0,15)
rownames(data.vsnrma) = tr

#read in ctas

ctas = as.character(data.breast[,1])

ctas.unique = unique(ctas)

ind = which(ensembl.symbols%in%ctas.unique)

ctas.symbols=ensembl.symbols[ind]

ctas.transcripts = names(ctas.symbols)

ind1=which(rownames(data.vsnrma)%in%ctas.transcripts)



tra.transcripts = rownames(data.vsnrma)[ind1]
tra.symbols = ctas.symbols[tra.transcripts]

data.tra = (data.vsnrma)[ind1,]

data.tra.table = cbind(data.tra,tra.symbols)

write.csv(data.cta.table,file = "data.tra.table.csv")

data.tra2 = data.tra.table.new[1,2]
rownames(data.tra2) = daza.tra.table.new[,1]

tra.symbols2 = data.tra.table.new[,2]

##barplot([1,],col="red",cex.names=0.6,main = paste (cta.transcripts)

boxplot(t(data.breast),col=rainbow(100),main = "TRAs in breast cancer ", cex.axis=0.4, names = tra.symbols2)

box.plot = paste("boxplot_breast_cancer_TRAs_",".eps",sep="")
dev.copy2eps(file="box.plot")





