---
title: "Project"
author: "Andreas Breu√ü, Anastassia Fink, Leon Hornich, Yuan Sun"
date: "19 5 2021"
output: html_document
---

---
## R Markdown

```{r}
knitr::opts_chunk$set(echo = TRUE)
```

# define working directory
```{r setup, include=FALSE}


wd="C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio"


setwd(paste0(wd,"/tra/"))


```


# libraries

```{r}
 
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf, lib.loc = "C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/Packages/")
library(hgu133plus2hsenstprobe, lib.loc ="C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/Packages/")
library(hexbin)
library(rstudioapi)
library(tidyverse)
```



# Read in .CEL files
```{r}
#Kontrolle
#ind = which (new%in%c ()) 
##dev.copy2eps(file="QC_.....eps")
##setwd("C:/Users/metho/iCloudDrive/Studium/4.FS/Bioinfo/RStudio/plots/")
#data.norm=vsnrma(data)

setwd(paste0(wd, "/rawdata/GSE149507 lung cancer"))

data = ReadAffy()
data@cdfName<-"HGU133PLUS2_Hs_ENST"

setwd(paste0(wd,"/sessions/rda"))

lungcancer_matrix <- as.data.frame(exprs(data))

save.image(file="rawdata.lungcancer.pdf")
image(data, col = rainbow(100, start = 0, end = 0.75)[100:1])



```


# overview
```{r}

#microarrays
dim(lungcancer_matrix)[2]

# names of the samples
colnames(data)
substr(colnames(lungcancer_matrix), 1,10)

# gene count
dim(lungcancer_matrix)[1]

# probe set names (or Affymetrix IDs) - are changed when changing cdfName
featureNames(data)[1:10]

#expression table
head(lungcancer_matrix)

```

#```{r}

setwd(paste0(wd,"/plots"))

for (i in 1:24){
  image(data[,i], col = rainbow(100, start = 0, end = 0.75)[100:1])
  file.name = paste0("lungcancer_", str_remove(colnames(lungcancer_matrix)[1],".CEL"),".pdf")
  # dev.copy2pdf(file = file.name)
}

```


# cleanup NA and/or space
```{r}
 
apply(lungcancer_matrix, 2, function(x){sum(is.na(x))})
apply(lungcancer_matrix, 2, function(x){sum(x == 0)})
sum(sapply(rownames(lungcancer_matrix), function(x){sum(x == "")}))

``````{r}

data.norm <- vsnrma(data)
data_vsnrma_matrix <- exprs(data.norm)

head(data.norm)
head(data_vsnrma_matrix)

#save

setwd(paste0(wd,"/sessions/rda"))

save.image(file="normalized_data..pdf")

```


# Plots
```{r}

#meanSdPlot
meanSdPlot(data.norm)

setwd(paste0(wd,"/plots/"))
dev.copy2eps(file="meanSdPlot_lungcancer_vsnrma_normalized.eps")



# Boxplot before normalization

par(las = 2) #axis labels 

par(mai = c(0.5,1,0.5,0.1))

boxplot(data, names = substr(colnames(lungcancer_matrix), 1,12),
        col = rainbow(24), cex.axis=0.6, 
        main="sample expression Lungcancer", 
        horizontal = FALSE)


setwd(paste0(wd,"/plots/"))
dev.copy2eps(file="boxplot_lungcancer_rawdata.pdf")



#Boxplot normalized data

par(las = 2) #axis labels 

par(mai = c(0.5,1,0.5,0.1))

boxplot(data_vsnrma_matrix, names = substr(colnames(lungcancer_matrix), 1,12),
        col = rainbow(24), cex.axis=0.6, 
        main="sample expression Lungcancer", 
        horizontal = FALSE)


setwd(paste0(wd,"/plots/"))
dev.copy2eps(file="boxplot_lungcancer_normalized.pdf")


```



# RNA degeneration plot
```{r}

rnadeg.raw = AffyRNAdeg(data)

#shift & scale
plotAffyRNAdeg(rnadeg.raw, col=rainbow(24), transform = "shift.scale")
title(sub = "lungcancer rawdata")


setwd(paste0(wd, "/plots/" ,sep = "/"))
dev.copy2pdf(file = "lungcancer_rnadeg_rawdata.pdf")


#shift
plotAffyRNAdeg(rnadeg.raw, col = rainbow(24), transform = "shift.only")
title(sub = "lungcancer rawdata")


setwd(paste0(wd, "/plots/" ,sep = "/"))
dev.copy2pdf(file = "lungcancer_rawdata.pdf")

```



# Scatterplot

```{r}

setwd(paste0(wd,"/plots/"))

for (i in 1:12){
  plot(exprs(data.norm)[,i], exprs(data.norm)[,i+1], pch = ".",
       xlab = substr(colnames(lungcancer_matrix), 1,10)[i],
       ylab = substr(colnames(lungcancer_matrix), 1,10)[i+1])
  abline(0,1,col="red")
  
  title(main = paste("Scatterplot of probe", 
                     substr(colnames(lungcancer_matrix), 1,10)[i],
                     "and",
                     substr(colnames(lungcancer_matrix), 1,10)[i+1],
                     sep = " ", collapse = NULL))
  
  file.name = paste("diabetes_scatterplot_",
                    substr(colnames(lungcancer_matrix), 1,10)[i],
                     "_",
                    substr(colnames(lungcancer_matrix), 1,10)[i+1],
                    ".pdf",
                    sep = "")
  dev.copy2pdf(file = file.name)
}


```


# Extract expression values
```{r}

dataExprs = exprs(data.norm)
head(dataExprs)
dim(dataExprs)

```

# ensemble IDs
```{r}

dataExprs = dataExprs[grepl("ENST", rownames(dataExprs)), ]
head(dataExprs)
dim(dataExprs)

```

# remove x_at suffix
```{r}

rownames(dataExprs) <- unlist(lapply(strsplit(rownames(dataExprs), split = "\\."), "[", 1))
head(dataExprs)

```

# ensemble IDs and gene symbols
```{r}
setwd(paste0(wd, "/tables"))
ensembl_df = read.csv("ensembl.103.txt", sep = ",")
head(ensembl_df)

transcriptIDs = as.character(ensembl_df[, "Transcript.stable.ID"])
geneSymbol = as.character(ensembl_df[, "HGNC.symbol"])
names(geneSymbol) = transcriptIDs
head(geneSymbol)
```

# Replace ensemble IDs with HGNC symbol
```{r}
chipIDs = rownames(dataExprs)
noMatchIDs = chipIDs[!chipIDs %in% transcriptIDs]
length(noMatchIDs)

newChipIDs = chipIDs[chipIDs %in% transcriptIDs]
dataExprs = dataExprs[newChipIDs, ]
dim(dataExprs)

symbol = geneSymbol[newChipIDs]
rownames(dataExprs) = as.character(symbol)
head(dataExprs)
```

# TRA input / experimental stage!!
```{r}

setwd(paste0(wd,"/tra/"))

human1 =read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv",sep=",")
human2 =read.csv(file="tra.2014.human.5x.table.tsv",sep="\t")
human3 =read.csv(file="tra.2014.human.roth.5x.table.tsv",sep="\t")
human4 =read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")


x = 1

tiss1=human1[,11]
tiss2=human2[,11]
tiss3=human3[,11]
tiss4=human4[,10]

ind1=which(tiss1=="thyroid")
ind2=which(tiss2=="thyroid")
ind3=which(tiss3=="thyroid")
ind4=which(tiss4=="thyroid")

TRA1.symbol=human1[,6]
TRA2.symbol=human2[,3]
TRA3.symbol=human3[,3]
TRA4.symbol=human4[,3]

lung.TRA1=TRA1.symbol[ind1]
lung.TRA2=TRA2.symbol[ind2]
lung.TRA3=TRA3.symbol[ind3]
lung.TRA4=TRA4.symbol[ind4]

#show(ind)

lung.TRA <- c(lung.TRA1, lung.TRA2, lung.TRA3, lung.TRA4)

show(lung.TRA)

which(lung.TRA =="CSNA")
which(lung.TRA =="WAP")
which(lung.TRA =="ELF5")
which(lung.TRA =="LALBA")
which(lung.TRA =="ODAM")
which(lung.TRA =="SULT1D1")



#head(human1)
#head(human2)
#head(human3)
#head(human4)

```

# Genes of interest

```{r}



ind=which(toupper(human1) %in% as.character(symbol))

dataExprs.sub=dataExprs[ind,]

dataExprs.sub

#show(ind)

```



#Analysis

###```

par(las=2)

boxplot(t(data.sub),col=rainbow(length(rownames(data.sub))),main="gene expression of thyroid-specific genes in thyroid cancer",cex.axis=0.8)

#sort alphabetically

order.vector=sort(colnames(t(data.sub)),index.return=T)$ix

boxplot(t(data.sub)[,order.vector],col=rainbow(length(rownames(data.sub))),main="gene expression of thyroid-specific genes in thyroid cancer",cex.axis=0.8)








